benches/encoding_primitives.rs

```diff
@@ -99,7 +99,7 @@ pub fn criterion_benchmark(c: &mut Criterion) {
         b.iter(|| {
             let mut encoded_length: usize = 0;
             for value in &unsigned_values {
-                encoded_length += black_box(FlexUInt::write_u64(&mut output, *value).unwrap());
+                encoded_length += black_box(FlexUInt::write(&mut output, *value).unwrap());
```

Comment:
🗺️ The `FlexUInt::write` method can now accept any int size. I ran the `encoding_primitives` benchmark and confirmed that this is the same or faster than the `u64`-specific method in all cases.